### ISSUE TITLE
Angular2 (>=rc7) dependency warning: "Critical dependency: the request of a dependency is an expression"

### ISSUE THREAD

**Issue by sbolel** [r-park/todo-angular2-firebase#96]
### Steps to reproduce and a minimal demo of the problem

- `git fetch --all && git checkout chore/update-angular2-final-release`
- `npm run build`
- observe webpack warnings

### Current behavior

- webpack outputs warnings:

    ```
    WARNING in ./~/@angular/core/src/linker/system_js_ng_module_factory_loader.js
    45:15 Critical dependency: the request of a dependency is an expression
    
    WARNING in ./~/@angular/core/src/linker/system_js_ng_module_factory_loader.js
    57:15 Critical dependency: the request of a dependency is an expression  
    ```

### Expected/desired behavior

- no warnings

### Solution

_UNTESTED_

`webpack.config.js`

```js
module.exports = {
  plugins: [
      new webpack.ContextReplacementPlugin(
        // The (\\|\/) piece accounts for path separators in *nix and Windows
        /angular(\\|\/)core(\\|\/)(esm(\\|\/)src|src)(\\|\/)linker/,
        root('./src') // location of your src
      )
  ]
}
function root(__path) {
  return path.join(__dirname, __path);
}
```

### Other information & resources

- Bug introduced in Angular2 starting with RC7, reported in [angular/angular](https://github.com/angular/angular)
  - [issues/11580](https://github.com/angular/angular/issues/11580)
- Issue also reported in [AngularClass/angular2-webpack-starter](https://github.com/AngularClass/angular2-webpack-starter)
  - [issues/993](https://github.com/AngularClass/angular2-webpack-starter/issues/993)
  - fixed by [pull/998](https://github.com/AngularClass/angular2-webpack-starter/pull/998)

**Comment by sbolel** [comment]
Just tested solution and its working. will add that to PR #95

**Comment by sbolel** [comment]
Fix pushed to #95
